### DISCUSSION TITLE
How to do shared tauri::command definitions for frontend and backend?

### DISCUSSION THREAD

**Discussion by Tsuguri** [tauri-apps/tauri#11093] [Q&A] [answered]
Hi!
I'm using tauri (v2 rc) with Rust frontend (leptos, but specific framework rather not relevant) and will have quite a lot of tauri::commands in my project. Calling them by name from the UI rubs me the wrong way. If I rename one of the functions but forget to change usages on UI side I will not get any compiler error, I'll be waiting for someone to hit this particular call that was missed/misspelled. I don't like that.

Closest I came up with is having all invoke() calls wrapped in functions in one place and never to call invoke() directly. That still poses a risk of missing something. 

I don't have an idea for a proper solution either. I guess I could have a macro to generate calling function (wrapper around invoke) and called function (marked with tauri::command) from one definition but I would have to do it in a way that allows me to NOT have my commands implementations in there.

I hope my description is understandable. If anyone have this problems solved in any way I'd love to get some pointers :)

**Comment by FabianLars** [comment, marked as answer]
check out this comment https://github.com/tauri-apps/tauri/issues/1514#issuecomment-1619337304 it's the best summary there currently is imo

**Reply by Tsuguri** [reply]
Thank you very much. tauri-specta seems to be more active and generates TS/JS schema from the code - doesn't support rust frontend.
tauri-bindgen integrates with tauri routing on the other hand and seems to make mixing sync and async commands... weird. I guess I'll still try to figure out something else myself.
